docs/source/user_guide_rewrite.md

```diff
@@ -0,0 +1,448 @@
+
+### Conditional Independence Tests
+
+*[Pure copy of existing documentation]*
+
+Assuming we have the following data:
+
+
+```
+>>> import numpy as np, pandas as pd
+*>>>*
+>>> X = np.random.normal(loc=0, scale=1, size=1000)
+>>> Y = 2 * X + np.random.normal(loc=0, scale=1, size=1000)
+>>> Z = 3 * Y + np.random.normal(loc=0, scale=1, size=1000)
+>>> data = pd.DataFrame(data=dict(X=X, Y=Y, Z=Z))
+```
+
+To test whether X is conditionally independent of Z given Y using the [kernel dependence measure](https://papers.nips.cc/paper/3201-a-kernel-statistical-test-of-independence.pdf), all you need to do is:
+
+
+```
+>>> import dowhy.gcm as gcm
+>>>
+>>> *# Null hypothesis: x is independent of y given z*
+>>> p_value = gcm.independence_test(X, Z, conditioned_on=Y)
+>>> p_value
+*0.48386151342564865*
+```
+
+If we define a threshold of 0.05 (as is often done as a good default), and the p-value is clearly above this, it says X and Z are indeed independent when we condition on Y. This is what we would expect, given that we generated the data using the causal graph X→Y→Z, where Z is conditionally independent of X given Y.
+To test whether X is independent of Z (*without* conditioning on Y), we can use the same function without the third argument.
+
+
+```
+>>> *# Null hypothesis: x is independent of y*
+>>> p_value = gcm.independence_test(X, Z)
+>>> p_value
+*0.0*
+```
+
+Again, we can define a threshold of 0.05, but this time the p-value is clearly below this threshold. This says X and Z *are* dependent on each other. Again, this is what we would expect, since Z is dependent on Y and Y is dependent on X, but we don’t condition on Y.
+
+### Refute causal structure
+
+*[Pure copy of existing documentation]*
+
+Testing the causal graph against data:
+
+```
+>>> rejection_result, _ = gcm.reject_causal_structure(causal_graph, data)
+>>> if rejection_result == gcm.RejectionResult.REJECTED:
+>>>  print("Do not continue. We advice revising the causal graph.")
+```
+
+*[TODO (kaibud or bloebp): this chapter should provide more content]*
+
+### Direct arrow strength
+
+*[Pure copy of existing documentation]*
+
+By quantifying the strength of an arrow, we answer the question:
+
+>How strong is the causal influence from a cause to its direct effect?
+
+#### How to use it
+
+To see how the method works, let us generate some data.
+
+```
+>>> import numpy as np, pandas as pd, networkx as nx
+>>> from dowhy import gcm
+>>> np.random.seed(10)  *# to reproduce these results*
+```
+
+```
+>>> Z = np.random.normal(loc=0, scale=1, size=1000)
+>>> X = 2*Z + np.random.normal(loc=0, scale=1, size=1000)
+>>> Y = 3*X + 4*Z + np.random.normal(loc=0, scale=1, size=1000)
+>>> data = pd.DataFrame(dict(X=X, Y=Y, Z=Z))
+```
+
+Next, we will model cause-effect relationships as a probabilistic causal model and fit it to the data.
+
+```
+>>> causal_model = gcm.ProbabilisticCausalModel(nx.DiGraph([('Z', 'Y'), ('Z', 'X'), ('X', 'Y')]))
```

Comment:
earlier in the documentation, you had used SCM. will be good to have a visualization on when to use which one, somewhere in the docs.

Reply:
@bloebp I'll probably need you help here.

Reply:
Good point, its probably a bit over-engineered as well currently. Will take a look/prepare a graphic.

Reply:
Added a dedicated section: docs/source/user_guide/miscellaneous/graphical_causal_models.rst